lib/internal/readline/eventsToAsyncIteratorFactory.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+const {
+  Promise,
+  SymbolAsyncIterator,
+} = primordials;
+
+const PAUSE_THRESHOLD = 1024;
+const RESUME_THRESHOLD = 1;
+const ITEM_EVENTS = ['data'];
+const CLOSE_EVENTS = ['close', 'end'];
+const ERROR_EVENTS = ['error'];
+
+let FixedQueue;
+let setImmediate;
+function waitNext(emitter, next, events) {
+  return new Promise((resolve, reject) => {
+    const resolveNext = () => {
+      events.forEach((event) => emitter.off(event, resolveNext));
+      setImmediate(() => {
+        try {
+          resolve(next());
+        } catch (promiseError) {
+          reject(promiseError);
+        }
+      });
+    };
+    events.forEach((event) => emitter.once(event, resolveNext));
```

Comment:
Avoid forEach loops, they are slower than for(;;) or for..of alternatives.

Reply:
Yes, we want to use `for(;;)` loops.

Reply:
@mcollina wait what? I thought they have been reduced to regular `for` loops for a few years now?

Reply:
@mcollina I've already changed it and I'm not going back, but now I'm curious about what @benjamingr said lol